### DISCUSSION TITLE
Need help with scrollbar not reaching the last line inside a ContentSwitcher widget

### DISCUSSION THREAD

**Discussion by learnbyexample** [Textualize/textual#3867] [Q&A] [answered]
Consider this sample program:

```python
from textual.app import App
from textual.widgets import Label, Button, Footer, ContentSwitcher
from textual.containers import VerticalScroll, Horizontal

class TestApp(App):

    CSS = '''
    Screen {
        overflow-y: hidden;
    }

    Button {
        max-height: 1;
        border: none;
        width: 1fr;
    }

    .horz {
        height: 1;
        align: center middle;
    }

    .vert {
        align: center middle;
    }
    '''

    def compose(self):
        with Horizontal(classes='horz'):
            yield Button('Test App')
        with ContentSwitcher(initial='foo'):  
            with VerticalScroll(id='foo', classes='vert'):
                yield Label('1 2 3 4 5 6 7 8 9 10 11 12 13'.replace(' ', '\n'))
        yield Footer()

    def on_mount(self):
        self.dark = False

if __name__ == "__main__":
    app = TestApp()
    app.run()
```

I've tried to create a minimal working example from my actual use case. I have a `Horizontal` container at the top, `ContentSwitcher` in the middle and `Footer` at the bottom. The problem I'm facing is that the scrollbar in the middle doesn't reach the last line as shown below:

![scroll_1](https://github.com/Textualize/textual/assets/17766317/bc5c4b44-02eb-433b-a8b6-6d6868f154b6)

If I remove `overflow-y: hidden`, then I do get the last line:

![scroll_2](https://github.com/Textualize/textual/assets/17766317/75be78b5-36d4-4761-a756-39909eb6bfbd)

But I don't want that extra entire app level scrollbar. I only need it for the `ContentSwitcher`. It's been a while since I worked on Textual, so I haven't really caught up to the latest changes. This set up used to work on version `0.32.0` (if you try the above sample program on this version, there's no missing line at the end).

**Comment by willmcgugan** [comment, marked as answer]
The ContentSwitcher has height auto by default, which means it will grow with the content. That's what may have changed since 0.32.0.

To make the ContentSwitcher only take up the remaining space in its container, you can set its height to `1fr`. I think this will give you the effect you are looking for:

```python
from textual.app import App
from textual.widgets import Label, Button, Footer, ContentSwitcher
from textual.containers import VerticalScroll, Horizontal


class TestApp(App):
    CSS = """
    Screen {
        overflow-y: hidden;
    }

    Button {
        max-height: 1;
        border: none;
        width: 1fr;
    }

    .horz {
        height: 1;
        align: center middle;
    }

    .vert {
        align: center middle;
    }

    ContentSwitcher {
        height: 1fr;
    }

    """

    def compose(self):
        with Horizontal(classes="horz"):
            yield Button("Test App")
        with ContentSwitcher(initial="foo"):
            with VerticalScroll(id="foo", classes="vert"):
                yield Label("1 2 3 4 5 6 7 8 9 10 11 12 13".replace(" ", "\n"))
        yield Footer()

    def on_mount(self):
        self.dark = False


if __name__ == "__main__":
    app = TestApp()
    app.run()from textual.app import App

```

**Reply by learnbyexample** [reply]
Thanks, that worked for my actual app too.
